### PR TITLE
cuBLAS doc + error if -ngl > 0 and no cuBLAS

### PR DESCRIPTION
**Pull request by JohannesGaessler**
This PR adds documentation for how to use the GPU accelerated token generation that I implemented. Also, if llama.cpp was compiled without cuBLAS and the user then tries to run llama.cpp with 1 or more gpu layers an arror is raised and informs the user that they need to compile with cuBLAS (this was a point of failure for multiple people).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -292,6 +292,8 @@ Building the program with BLAS support may lead to some performance improvements
     cmake --build . --config Release
     ```
 
+Prompt processing will automatically be GPU accelerated. To enable token generation acceleration use the `-ngl` or `--n-gpu-layers` argument and specify how many layers should be offloaded to the GPU. A higher value will enable more GPU acceleration but also increase VRAM usage. Maximum effective values: 33 for 7b, 41 for 13b, 61 for 33b, 81 for 65b. Multi-GPU setups and iGPUs are currently not supported.
```
Comment (Green-Sky) [inline README.md:295]: are those values right? afaik 7B has 32 layers.

Comment (JohannesGaessler) [reply README.md:295]: If the value of GPU layers is higher than the number of repeating layers then the non-repeating layers are also getting offloaded so 33 is still giving you an improvement for 7b compared to 32.

File: llama.cpp [hunk 2]
```diff
@@ -1054,7 +1054,11 @@ static void llama_model_load_internal(
         fprintf(stderr, "%s: [cublas] total VRAM used: %zu MB\n", __func__, vram_total / 1024 / 1024);
     }
 #else
-    (void) n_gpu_layers;
+    if (n_gpu_layers > 0) {
+        fprintf(stderr, "Error: llama.cpp was compiled without cuBLAS. "
+            "It is not possible to offload the requested %d layers onto the GPU.\n", n_gpu_layers);
+        exit(1);
+    }
```
Comment (ggerganov) [inline llama.cpp:1061]:
```suggestion
    if (n_gpu_layers > 0) {
        throw format(" ... explain error ...");
    }
```

**Review by ggerganov — APPROVED** [review]

**Comment by Green-Sky** [comment]
Looks like a temporary issue. I'm rerunning the failed jobs, should fix it.

**Comment by Green-Sky** [comment]
click on the jobs details and then in the upper right there is a re-run button. If something fails, you can also select re-run failed. :)

**Pull request closed without merging**